### DISCUSSION TITLE
캠스터디 서비스에서 실시간 참여자 수 및 비디오 스트림 관리 시 발생하는 리렌더링 문제 해결을 위한 최적화 전략 연구

### DISCUSSION THREAD

**Discussion by yeji0214** [100-hours-a-week/discussions#32] [Deepdive]
## 상황
- 캠스터디 세션 동안 최대 5명의 참여자가 실시간으로 비디오 스트림을 송수신합니다. 참여자의 입장 및 퇴장, 비디오 스트림의 품질 변화(예: 네트워크 상태에 따른 해상도 조절), 실시간 채팅 메시지, 화면 공유 등의 다양한 상태가 매우 빈번하게 업데이트되고 있습니다.

## 문제
- 고빈도 상태 업데이트로 인해 리액트 컴포넌트들이 불필요하게 자주 리렌더링되고 있으며, 특히 참여자가 많아질수록 리렌더링으로 인한 성능 저하가 발생하게 됩니다. 이는 캠스터디 세션 도중 비디오 스트림의 끊김, UI 반응 속도 저하 등으로 이어져 사용자 경험에 부정적인 영향을 미칠 것으로 예상됩니다.

## 목표
- 실시간 상태 관리가 필요한 상황에서, 불필요한 리렌더링을 최소화하고 컴포넌트의 성능을 최적화하여 원활한 사용자 경험을 제공하는 것을 목표로 합니다.

**Comment by startupcodekr** [comment]
캠스터디 서비스에서 실시간 참여자 수 및 비디오 스트림 관리 시 발생하는 리렌더링 문제 해결을 위한 최적화 전략
